compiler/src/iree/compiler/Codegen/Common/TransformExtensions/TransformMatchers.cpp

```diff
@@ -0,0 +1,356 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/TransformExtensions/TransformMatchers.h"
+
+#include "mlir/Analysis/SliceAnalysis.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+
+using namespace mlir;
+using namespace mlir::iree_compiler;
+using namespace mlir::iree_compiler::IREE;
+
+//===---------------------------------------------------------------------===//
+// StructuredOpMatcher and friends.
+//===---------------------------------------------------------------------===//
+
+bool transform_dialect::StructuredOpMatcher::match(Operation *op) {
```

Comment:
This too. Lot of these seem like they are to be upstreamed.... If so can we land these in `llvm-external-projects`.

Reply:
Thomas proposed to move stuff as a followup to help distribute the load.
Leaving that to him, thanks Thomas!